src/cosp.F90

```diff
@@ -1832,10 +2930,113 @@ END SUBROUTINE COSP_INIT
   !%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
   ! SUBROUTINE cosp_cleanUp
   !%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
-  subroutine cosp_cleanUp()
+  subroutine cosp_cleanUp()  
     deallocate(vgrid_zl,vgrid_zu,vgrid_z,dz)
   end subroutine cosp_cleanUp
 
+  !%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+  ! SUBROUTINE compute_orbitmasks
+  !%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%  
+  subroutine compute_orbitmasks(Npoints,Nlocaltimes,localtimes,localtime_widths,      &
+                                lat,lon,month,day,hour,minute,swath_mask_out,Nswathed_out)
+
+    ! Inputs
+    integer,intent(in) :: &
+      Npoints,         &
+      Nlocaltimes
+
+    real(wp),dimension(Nlocaltimes),intent(in) :: &
+      localtimes,        &
+      localtime_widths
+
+    real(wp),dimension(Npoints),intent(in) :: &
+      lat,     &
+      lon,     &
+      month,   &
+      day,     &
+      hour,    &
+      minute
+
+    ! Output
+    logical,dimension(Npoints),intent(out) :: &
+      swath_mask_out    ! Mask of reals over all gridcells
+    integer,intent(out) :: &
+      Nswathed_out
+
+    ! Local variables
+    integer :: i   ! iterator
+
+    real(wp),parameter                     :: &
+      pi = 4.D0*DATAN(1.D0),  &  ! yum
```

Comment:
Pi is already defined in the cosp_constants module. Reference that value instead of redefining.